core/modules/cart/store/actions.ts

```diff
@@ -141,7 +141,6 @@ const actions: ActionTree<CartState, RootState> = {
     if (options.recreateAndSyncCart && getters.isCartSyncEnabled) {
       await commit(types.CART_LOAD_CART_SERVER_TOKEN, null)
       await commit(types.CART_SET_ITEMS_HASH, null)
-      await dispatch('connect', { guestCart: !config.orders.directBackendSync }) // guest cart when not using directBackendSync because when the order hasn't been passed to Magento yet it will repopulate your cart
```

Comment:
How do we handle this directBackendSync = false? It’s when orders are not directly passed to the server but rather queued in local storage (offline mode). In that case the cart should be recreated as guest cart bc you can’t be sure that the order has been already passed to the server so the authorized user quote can still contain the products so the effect will be they “return” to the cart after placing an order

Reply:
It sounds complicated :sweat_smile: Why not always clear cart after order? No matter if it's send to backend or not. We don't need to send cart token with order because it will be created in api https://github.com/DivanteLtd/vue-storefront-api/blob/master/src/platform/magento2/o2m.js#L65

So maybe let's make it even simpler:
```
  async clear ({ commit, dispatch, getters }, options = { recreateAndSyncCart: true }) {
    await commit(types.CART_LOAD_CART, [])
    await commit(types.CART_LOAD_CART_SERVER_TOKEN, null)
    await commit(types.CART_SET_ITEMS_HASH, null)
  },
```
If we always clear cart then there is no problem with returning products back to cart. :thinking: Do we need to keep cart token after order, even if it's not send?

Reply:
Ok, I make it as wrote before. So now we will remove cart always after an order.